sbi/inference/snpe/snpe_base.py

```diff
@@ -88,42 +84,40 @@ def __init__(
         # create the deep neural density estimator
         if density_estimator is None:
             density_estimator = utils.posterior_nn(
-                model="maf", prior=self._prior, context=self._true_observation,
+                model="maf", prior=self._prior, x_o=self._x_o,
             )
         # create the neural posterior which can sample(), log_prob()
         self._neural_posterior = Posterior(
             algorithm_family="snpe",
             neural_net=density_estimator,
             prior=prior,
-            context=self._true_observation,
+            x_o=self._x_o,
             sample_with_mcmc=sample_with_mcmc,
             mcmc_method=mcmc_method,
             get_potential_function=PotentialFunctionProvider(),
         )
 
-        # obtain z-score for observations and define embedding net
-        if self.z_score_obs:
-            self.obs_mean = torch.mean(self.pilot_observations, dim=0)
-            self.obs_std = torch.std(self.pilot_observations, dim=0)
+        # obtain z-score for data x and define embedding net
+        if self._z_score_x:
+            self.x_mean = torch.mean(self.pilot_x, dim=0)
+            self.x_std = torch.std(self.pilot_x, dim=0)
         else:
-            self.obs_mean = torch.zeros(self._true_observation.shape)
-            self.obs_std = torch.ones(self._true_observation.shape)
+            self.x_mean = torch.zeros(self._x_o.shape)
+            self.x_std = torch.ones(self._x_o.shape)
 
         # new embedding_net contains z-scoring
         if not isinstance(self._neural_posterior.neural_net, MultivariateGaussianMDN):
             embedding = nn.Sequential(
-                utils.Normalize(self.obs_mean, self.obs_std),
+                utils.Standardize(self.x_mean, self.x_std),
                 self._neural_posterior.neural_net._embedding_net,
             )
             self._neural_posterior.set_embedding_net(embedding)
-        elif z_score_obs:
-            warnings.warn("z-scoring of observation not implemented for MDNs")
+        elif z_score_x:
+            warnings.warn("z-scoring of data x not implemented for MDNs")
 
         # calibration kernels proposed in Lueckmann, Goncalves et al 2017
         if calibration_kernel is None:
-            self.calibration_kernel = lambda context_input: torch.ones(
-                [len(context_input)]
-            )
+            self.calibration_kernel = lambda x: torch.ones([len(x)])
```

Comment:
I think we just want the number of datapoints `x` in the batch (i.e. batch_size), so the leading dimension should be fine.

Reply:
Just for my understanding: this is the default or the case when no calibration kernel is provided, i.e., the kernel returns 1 for every `x` passed to it so that it does not affect the loss, because it is just a factor in front of the importance weights or the log probs in the loss?